vmupdate/agent/entrypoint.py

```diff
@@ -74,7 +74,14 @@ def get_package_manager(os_data, log, log_handler, log_level, no_progress):
             from source.apt.apt_cli import APTCLI as PackageManager
     elif os_data["os_family"] == "RedHat":
         try:
-            from source.dnf.dnf_api import DNF as PackageManager
+            version = int(os_data["release"].split(".")[0])
+        except ValueError:
+            version = 99  # fedora changed its version
+        try:
+            if version < 41:
+                from source.dnf.dnf_api import DNF as PackageManager
+            else:
+                from source.dnf.dnf5_api import DNF as PackageManager
         except ImportError:
             log.warning("Failed to load dnf with progress bar. Use dnf cli.")
```

Comment:
I hit this case initially. Several things:
- I didn't got this message in GUI updater, only in log file, why?
- The "use dnf cli" suggest an action from the user, -> "using"?
- Maybe try fallback to dnf4 before fallback to CLI? Or could that cause issues?
- Should core-agent-linux pull in python3-libdnf5? Or maybe qubes-vm-recommended (in which case, it won't be present in minimal template)? With DNF4 it wasn't an issue, as it was a dependency of DNF itself...

Reply:
- Should be better now
- fixed
- you're right
- I think it should